crates/libm-test/src/mpfloat.rs

```diff
@@ -240,6 +240,19 @@ macro_rules! impl_op_for_ty {
                 }
             }
 
+            impl MpOp for crate::op::[<ilogb $suffix>]::Routine {
+                type MpTy = MpFloat;
+
+                fn new_mp() -> Self::MpTy {
+                    new_mpfloat::<Self::FTy>()
+                }
+
+                fn run(this: &mut Self::MpTy, input: Self::RustArgs) -> Self::RustRet {
+                    this.assign(input.0);
+                    this.get_exp().unwrap_or(i32::MIN)
```

Comment:
[`get_exp`](https://docs.rs/rug/latest/rug/struct.Float.html#method.get_exp) follows [`frexp`](https://en.cppreference.com/w/cpp/numeric/math/frexp)'s rules for determining which exponent to use, which is to pick `v = m * 2^e` where `0.5 <= |m| < 1.0`. [`ilogb`](https://en.cppreference.com/w/cpp/numeric/math/ilogb) by contrast uses `e = floor(log2(|v|))` (without rounding errors) which is equivalent to `1.0 <= |m| < 2.0`.
so, you need `get_exp().map(|v| v - 1)`